kubric/core/assets.py

```diff
@@ -173,11 +171,9 @@ def __repr__(self):
     else:
       return f"<{self.uid}>"
 
-
 class Undefined:
   pass
 
-
```

Comment:
@Qwlouse is there a strong need to have "Undefined" and "UndefinedAsset"?
Why can't we merge the two?

I see they are used in different parts of the code in different way.

Reply:
I think I wanted to avoid "diamond" patterns in the inheritance structure, where e.g.  

`Asset` >> `Material` >> `UndefinedMaterial` << `UndefinedAsset` << `Asset`

But I'm not sure if that is even a problem, or if Python can simply handle this well.

Reply:
Ok, so I've read up on this a bit, and I think we can merge `Undefined` and `UndefinedAsset` as long as we remember to properly use `super()` when calling base-class methods from overridden ones.